include/mbedtls/pk.h

```diff
@@ -234,10 +234,17 @@ typedef struct mbedtls_pk_info_t mbedtls_pk_info_t;
 
 /**
  * \brief           Public key container
+ *
+ * \note            The opaque_id is guarded by MBEDTLS_PSA_CRYPTO_C and not
+ *                  only by MBEDTLS_USE_PSA_CRYPTO because it can be used also
+ *                  in mbedtls_pk_sign_ext for RSA keys.
  */
 typedef struct mbedtls_pk_context {
     const mbedtls_pk_info_t *MBEDTLS_PRIVATE(pk_info);    /**< Public key information         */
     void *MBEDTLS_PRIVATE(pk_ctx);                        /**< Underlying public key context  */
+#if defined(MBEDTLS_PSA_CRYPTO_C)
+    mbedtls_svc_key_id_t MBEDTLS_PRIVATE(opaque_id);      /**< Key ID for opaque keys */
```

Comment:
Note: I'm wondering if this is the best name, considering we're going to use that for non-opaque keys pretty soon. Should we call it `prv_id` instead? For now the comment would still say "for opaque keys" and then when using it for non-opaque we'd just have to remove that part of the comment. Wdyt?

Reply:
Yes, I wasn't unsure as well for the naming. I went for `opaque_id` because it seemed slightly better for the purpose of this PR, but I agree that for a wider point of view it might be better to name it properly since now.